crates/services/txpool_v2/src/extracted_outputs.rs

```diff
@@ -88,6 +88,35 @@ impl ExtractedOutputs {
         }
     }
 
+    pub fn new_extracted_outputs<'a>(
+        &mut self,
+        outputs: impl Iterator<Item = &'a Output>,
+        tx_id: TxId,
+    ) {
+        for output in outputs {
+            match output {
+                Output::ContractCreated { contract_id, .. } => {
+                    self.contract_created.insert(*contract_id, tx_id);
+                }
+                Output::Coin {
+                    to,
+                    amount,
+                    asset_id,
+                } => {
+                    self.coins_created
+                        .entry(tx_id)
+                        .or_default()
+                        .insert(UtxoId::new(tx_id, 0), (*to, *amount, *asset_id));
+                }
+                Output::Contract { .. }
+                | Output::Change { .. }
+                | Output::Variable { .. } => {
```

Comment:
Resolved outputs only contain `Change` and `Variable` outputs=D

Reply:
BTW, we need to make a note to add an integration test to check that we can submit transaction that uses `Change` or `Variable` output form its parent after receiving a preconfiramtion